### PR TITLE
CLI switch to suppress internal API usages by internal plugins

### PR DESCRIPTION
**Pull request by novotnyr**
Add a `--suppress-internal-api-usages` switch to disable internal API usages reporting by internal plugins.

See #952 for original implementation, which did not provide CLI switch.

### CODE REVIEW

**Review by LChernigovskaya — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -149,11 +149,25 @@ This command is used to check one or more plugins against one or more IDEs ([opt
...
+        [-suppress-internal-api-usages no|internal-plugins]
 
 `<plugins>` is either `<plugin path>` or `'@<file>'` with a list of plugin paths to verify, separated by a newline.
 
 `<IDE>` is either a path to local IDE installation, or an IDE pattern (see bellow in the [common options](#common-options)) 
 
+#### Specific options
+
+* `-suppress-internal-api-usages` will suppress internal API usages from internal plugins.
+This option is used by JetBrains MarketPlace by default.
```
Comment (YannCebron) [inline README.md:161]: JetBrains Market<p>lace

"internal plugins" is misleading, it should be "plugins from JetBrains" (see also below)

the cmd-line option should also be renamed

File: intellij-plugin-verifier/verifier-cli/src/main/java/com/jetbrains/pluginverifier/tasks/checkPlugin/CheckPluginParams.kt [hunk 2]
```diff
@@ -31,4 +32,16 @@ class CheckPluginParams(
...
+}
+
+enum class InternalApiVerificationMode {
+  /**
+   * Full verification of internal API usages for all plugins.
+   */
+  FULL,
+
+  /**
+   * Internal plugins that use internal APIs will not be reported.
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-cli/src/main/java/com/jetbrains/pluginverifier/tasks/checkPlugin/CheckPluginParams.kt:44]: see above

**Review by YannCebron — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -149,11 +149,22 @@ This command is used to check one or more plugins against one or more IDEs ([opt
...
+        [-suppress-internal-api-usages no|jetbrains-plugins]
 
 `<plugins>` is either `<plugin path>` or `'@<file>'` with a list of plugin paths to verify, separated by a newline.
 
 `<IDE>` is either a path to local IDE installation, or an IDE pattern (see bellow in the [common options](#common-options)) 
 
+#### Specific options
+
+* `-suppress-internal-api-usages` will suppress internal API usages from JetBrains plugins.
+This option is used by JetBrains MarketPlace by default.
```
Comment (YannCebron) [inline README.md:161]: `JetBrains Market<p>lace`

File: intellij-plugin-verifier/verifier-cli/src/main/java/com/jetbrains/pluginverifier/tasks/checkPlugin/CheckPluginParams.kt (hunk 2 above)
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-cli/src/main/java/com/jetbrains/pluginverifier/tasks/checkPlugin/CheckPluginParams.kt:44]: still "internal plugins" in doc

**Review by YannCebron — CHANGES_REQUESTED** [review]

**Review by YannCebron — APPROVED** [review]

**Pull request merged**